src/api/posts.ts

```diff
@@ -0,0 +1,22 @@
+import { Post } from '../types/Post';
+import { request } from './api';
+
+export const getAllPosts = (endpoint?: string) => {
+  if (endpoint) {
+    return request(`/posts${endpoint}`);
+  }
+
+  return request('/posts');
+};
+
+export const getUserPosts = (userId?: number): Promise<Post[]> => {
+  if (!userId) {
+    return getAllPosts();
+  }
+
+  return getAllPosts(`/?userId=${userId}`);
+};
+
+export const getPostDetails = (postId: number): Promise<Post> => (
+  getAllPosts(`/${postId}`)
```

Comment:
```suggestion
  getAllPosts(`posts/${postId}`)
```
![image](https://user-images.githubusercontent.com/63306833/163808874-6e036228-8e9c-4401-8bb9-3c0815e4e0a9.png)